api/provider/aws/generic_load_balancer.go

```diff
@@ -0,0 +1,51 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/service/elb"
+	alb "github.com/aws/aws-sdk-go/service/elbv2"
+)
+
+func newGenericLoadBalancer(CLB *elb.LoadBalancerDescription, ALB *alb.LoadBalancer) *genericLoadBalancer {
+	if CLB != nil {
+		return &genericLoadBalancer{
+			CLB:   CLB,
+			isCLB: true,
+		}
+	}
+
+	if ALB != nil {
+		return &genericLoadBalancer{
+			ALB:   ALB,
+			isALB: true,
+		}
+	}
+
+	return nil
+}
+
+type genericLoadBalancer struct {
```

Comment:
[optional] This might be a more scalable approach:
```
type LoadBalancer interface {
    Scheme() string
    DNSName() string
}

type ALB alb.LoadBalancer 

func (a ALB) Scheme() string {
    // this is a name collision, but you get the idea
    return a.Scheme
}

func (a ALB) DNSName() string {
    return a.DNSName
}

type CLB elb.LoadBalancerDescription
...
```

Reply:
I still need access to the underlying ELB & ALB types which map to elb.LoadBalancerDescription & alb.LoadBalancer respectively. I suppose I could have something like this if I wanted to go via the interface route but not sure how it would be beneficial

```go
type LoadBalancer interface {
	Scheme() string
	DNSName() string
	ClassicLoadBalancer
	ApplicationLoadBalancer
}

type ClassicLoadBalancer interface {
	ELB() *elb.LoadBalancerDescription
}

type ApplicationLoadBalancer interface {
	ALB() *alb.LoadBalancer
}

```

Reply:
> I still need access to the underlying ELB & ALB types 

Why? Isn't just for conversion to/from layer0 models/aws models? If so, why not have the conversion(s) in the interface?